### PR TITLE
Alpha build setup

### PR DESCRIPTION
**Pull request by ifbarrera**
# What

Adds necessary jobs/steps to allow for automatic KickAlpha build uploads.

# Why

Preparation steps for automatic alpha builds.

# How

Merges into master will trigger the creation of a new branch `alpha-dist-<COMMIT>` on the private remote. Any builds to `alpha-dist-.*` branches will run a job that runs the same steps as the `beta` job, but calls `fastlane alpha_match_gym_hockey` instead of `fastlane beta_match_gym_hockey`. The `alpha_match_gym_hockey` lane will build the KickAlpha app and upload it to Hockey.

### CODE REVIEW

**Comment by ifbarrera** [comment]
[Automatic Alpha Build](https://trello.com/c/AQeq2Bvk/686-automatic-alpha-build)

File: circle.yml [hunk 1]
```diff
@@ -184,7 +184,71 @@ jobs:
       - store_artifacts:
           path: /tmp/xcode_raw.log
 
-  # Hockey
+  master:
```
Comment (justinswart) [inline circle.yml]: What's the `master` job for?

File: Makefile [hunk 2]
```diff
@@ -92,12 +92,21 @@ deploy:
...
 	@echo "Deploying private/alpha-dist..."
 
-	@git branch -f alpha-dist private/alpha-dist
+	@git branch -f alpha-dist private/$(BRANCH)
 	@git push -f private alpha-dist
 	@git branch -d alpha-dist
 
 	@echo "Deploy has been kicked off to CircleCI!"
 
+sync:
```
Comment (justinswart) [inline Makefile]: Also possibly not needed?

File: circle.yml [hunk 3]
```diff
@@ -292,6 +356,26 @@ workflows:
       - library-tests
       - livestream-tests
       - ksapi-tests
+      # - master:
```
Comment (justinswart) [inline circle.yml]: Guessing we won't need this if `master`'s not needed?

File: circle.yml (hunk 1 above)
Comment (justinswart) [reply circle.yml]: Oh I see now - your intention is that every time we push to `master` it should also release an alpha build 🤔

Ok, we can try it this way - there may be another way to inject the environment variable though if that doesn't work.

File: Makefile (hunk 2 above)
Comment (justinswart) [reply Makefile]: Answered my question second time around 😊

File: circle.yml (hunk 3 above)
Comment (justinswart) [reply circle.yml]: Answered my question second time around 😊

**Comment by justinswart** [comment]
@ifbarrera this seems good to me, received an alpha build. Guess all that's left is to rename the branch back to `master`?

**Pull request merged**